rql.go

```diff
@@ -134,14 +137,31 @@ func NewParser(c Config) (*Parser, error) {
 	}
 	p := &Parser{
 		Config: c,
-		fields: make(map[string]*field),
+		fields: make(map[string]*Field),
 	}
 	if err := p.init(); err != nil {
 		return nil, err
 	}
 	return p, nil
 }
 
+// Does not use config.Model, gets config from Fields)
+func NewParserF(c Config, fields []*Field) (*Parser, error) {
```

Comment:
Rename to: NewFieldsParser or add a new option as `Config.Fields`.
 
Also, an error should return if `c.Model` was set as the two are mutually exclusive.